schema/schemaspec/schemahcl/hcl.go

```diff
@@ -70,12 +69,69 @@ func (s *State) UnmarshalSpec(data []byte, v interface{}) error {
 	return s.Eval(data, v, nil)
 }
 
-// Eval implements the Evaluator interface.
-func (s *State) Eval(data []byte, v interface{}, input map[string]string) error {
-	spec, err := s.eval(data, input)
+// EvalFiles evaluates the files in the provided paths using the input variables and
+// populates v with the result.
+func (s *State) EvalFiles(paths []string, v interface{}, input map[string]string) error {
```

Comment:
I feel like this should take a slice of byte slices instead

Reply:
Hcl parser had nice diagnostics that includes file names

Reply:
A `[][]byte` for the file paths? Can you elaborate on why?